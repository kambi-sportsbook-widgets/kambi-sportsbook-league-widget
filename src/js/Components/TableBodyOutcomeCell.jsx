import React from 'react';
import { OutcomeButton } from 'kambi-widget-components';

const TableBodyOutcomeCell = ({ outcome, event }) => {
   return (
      <div className="l-flexbox l-flex-2 l-ml-6" data-item-attr="betOffer">
         <OutcomeButton outcome={outcome} event={event} />
      </div>
   );
};

TableBodyOutcomeCell.propTypes = {
   /**
    * Outcome entity
    */
   outcome: React.PropTypes.object.isRequired,

   /**
    * Optional event entity
    */
   event: React.PropTypes.object
};

export default TableBodyOutcomeCell;
