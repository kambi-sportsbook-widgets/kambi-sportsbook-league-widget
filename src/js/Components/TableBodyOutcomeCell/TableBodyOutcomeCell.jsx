import React from 'react';
import PropTypes from 'prop-types';
import styles from './TableBodyOutcomeCell.scss';

const TableBodyOutcomeCell = ({ children }) => {
   return (
      <td className={styles.general}>
         <div className={styles.wrapper}>
            {children}
         </div>
      </td>
   );
};

TableBodyOutcomeCell.propTypes = {
   /**
    * Inner element - OutcomeButton
    */
   children: PropTypes.node
};

export default TableBodyOutcomeCell;
