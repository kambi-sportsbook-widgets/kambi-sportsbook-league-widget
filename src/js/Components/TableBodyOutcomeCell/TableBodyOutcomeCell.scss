.general {
   width: 10%;
   padding: 0 0 0 3px; // makes the space between outcome buttons 3px

   .wrapper {
      display: flex;
   }
}

@media (max-width: 480px) {
   .general {
      width: 17%;
   }
}
