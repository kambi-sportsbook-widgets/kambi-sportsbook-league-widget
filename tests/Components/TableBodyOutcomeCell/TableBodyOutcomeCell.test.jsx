/* eslint-env jest */
import React, { Children } from 'react';
import TableBodyOutcomeCell from '../../../src/js/Components/TableBodyOutcomeCell/TableBodyOutcomeCell';
import ReactTestUtils from 'react-addons-test-utils';

const renderer = ReactTestUtils.createRenderer();

describe('TableBodyOutcomeCell DOM rendering', () => {

   it('renders correctly without children', () => {
      expect(renderer.render(
         <TableBodyOutcomeCell />
      )).toMatchSnapshot();
   });

   it('renders correctly with children', () => {
      expect(renderer.render(
         <TableBodyOutcomeCell>
            <div>Test</div>
         </TableBodyOutcomeCell>
      )).toMatchSnapshot();
   });

});
