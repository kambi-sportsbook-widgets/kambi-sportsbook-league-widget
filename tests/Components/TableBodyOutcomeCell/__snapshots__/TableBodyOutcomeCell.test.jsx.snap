// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyOutcomeCell DOM rendering renders correctly with children 1`] = `
<td
  className="general"
>
  <div
    className="wrapper"
  >
    <div>
      Test
    </div>
  </div>
</td>
`;

exports[`TableBodyOutcomeCell DOM rendering renders correctly without children 1`] = `
<td
  className="general"
>
  <div
    className="wrapper"
  />
</td>
`;
